airbyte-cdk/python/airbyte_cdk/test/utils/assertions.py

```diff
@@ -0,0 +1,23 @@
+import re
+
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput
+
+
+def is_in_logs(pattern: str, output: EntrypointOutput) -> bool:
+    return any(re.search(pattern, entry.log.message, flags=re.IGNORECASE) for entry in output.logs)
+
+
+def is_not_in_logs(pattern: str, output: EntrypointOutput) -> bool:
+    return not is_in_logs(pattern, output)
+
+
+def assert_good_read(output: EntrypointOutput, expected_records: int) -> None:
```

Comment:
nit: can we call `expected_records_count` to not confuse w/ our concept of expected records.